### PR TITLE
Adding missing javascript extension for grid - DO NOT MERGE

### PR DESCRIPTION
**Pull request by tomas862**
this Pr only demonstrates the usage of how you can add missing javascript from module to symfony controller when needed. According to this PR documentation will be created.

@matks - let me know if I can start adding docs according to this PR.

### CODE REVIEW

File: assets/js/admin/grid/customer/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import Grid from '../../../../../../../admin-dev/themes/new-theme/js/components/grid/grid';
```
Comment (mickaelandrieu) [inline assets/js/admin/grid/customer/index.js:1]: we agree that import is not possible in production :)

How can we now where are our core dependencies ?

Comment (matks) [reply assets/js/admin/grid/customer/index.js:1]: Looks like this is a pain point in Javascript
https://stackoverflow.com/questions/29168433/es6-variable-import-name-in-node-js
https://forums.meteor.com/t/how-to-use-import-require-with-dynamic-path/21955/15

Comment (tomas862) [reply assets/js/admin/grid/customer/index.js:1]: only two options I can see:

1. Move admin javascripts to `assets` folder in core ( the folder which name will never change)
2. Keep this approach -  the compiled result works fine even though other PrestaShop users using the module might have different folder name. If someone wants to develop the module then he will just change the imports and recompile hes own code.

**Comment by matks** [comment]
@tomas862 
1) I think we can merge your PR* and write documentation about it, that is a first good point 👍 

*Maybe without using babel 😅 we might want to make the example simpler in order for every kind of developer, both junior or senior, to understand it

2) I'm wondering if we could find another way to modify the main page
See this example:
```
import EmailSendingTest from './email-sending-test';
import SmtpConfigurationToggler from './smtp-configuration-toggler';
import Grid from '../../components/grid/grid';
import ReloadListActionExtension from '../../components/grid/extension/reload-list-extension';
import ...

const $ = window.$;

$(() => {
  const emailLogsGrid = new Grid('email_logs');

  emailLogsGrid.addExtension(new ReloadListActionExtension());
  emailLogsGrid.addExtension(new ExportToSqlManagerExtension());
  emailLogsGrid.addExtension(new FiltersResetExtension());
  emailLogsGrid.addExtension(...);

  const selector = '.my-selector';

  new EmailSendingTest(selector);
  new SmtpConfigurationToggler();
});
```
What if someone wants to modify the selector, because he modified the Twig template ? What if he wants to call some JS extension before SmtpConfigurationToggler or replace the SmtpConfigurationToggler by a custom one ?

I dont know whether this is a useful extension usecase 🤔 . 

If this indeed would be a good thing, then maybe some hooks like `addJSBefore` `addJSAfter` ?

File: assets/js/admin/grid/customer/index.js (hunk 1 above)
Comment (PierreRambaud) [reply assets/js/admin/grid/customer/index.js:1]: Or use library as it's defined in webpack configuration.
https://github.com/PrestaShop/PrestaShop/blob/develop/admin-dev/themes/new-theme/.webpack/common.js#L62

**Pull request closed without merging**